### ISSUE TITLE
Error validating master template generated via Creator tool

### ISSUE THREAD

**Issue by mfgonzalez** [Azure/azure-api-management-devops-resource-kit#301]
Hi!

I´m getting an error when validating the templates from the Creator tool. The error is the following:

`##[error]Deployment template validation failed: 'The value for the template parameter 'ApimServiceName' at line '1' and column '179' is not provided. Please see https://aka.ms/resource-manager-parameter-files for usage details.'.`

Here is the content of my master template:

`{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "ApimServiceName": {
      "type": "string",
      "metadata": {
        "description": "Name of the API Management"
      }
    },
    "LinkedTemplatesBaseUrl": {
      "type": "string",
      "metadata": {
        "description": "Base URL of the repository"
      }
    }
  },
  "resources": [
    {
      "properties": {
        "mode": "Incremental",
        "templateLink": {
          "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/gmsd1zu2amsplatfogene001-apiVersionSets.template.json')]",
          "contentVersion": "1.0.0.0"
        },
        "parameters": {
          "ApimServiceName": {
            "value": "[parameters('ApimServiceName')]"
          }
        }
      },
      "name": "versionSetTemplate",
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2018-01-01",
      "dependsOn": []
    },
    {
      "properties": {
        "mode": "Incremental",
        "templateLink": {
          "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/gmsd1zu2amsplatfogene001-products.template.json')]",
          "contentVersion": "1.0.0.0"
        },
        "parameters": {
          "ApimServiceName": {
            "value": "[parameters('ApimServiceName')]"
          }
        }
      },
      "name": "productsTemplate",
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2018-01-01",
      "dependsOn": []
    },
    {
      "properties": {
        "mode": "Incremental",
        "templateLink": {
          "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/gmsd1zu2amsplatfogene001-loggers.template.json')]",
          "contentVersion": "1.0.0.0"
        },
        "parameters": {
          "ApimServiceName": {
            "value": "[parameters('ApimServiceName')]"
          }
        }
      },
      "name": "loggersTemplate",
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2018-01-01",
      "dependsOn": []
    },
    {
      "properties": {
        "mode": "Incremental",
        "templateLink": {
          "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/gmsd1zu2amsplatfogene001-tags.template.json')]",
          "contentVersion": "1.0.0.0"
        },
        "parameters": {
          "ApimServiceName": {
            "value": "[parameters('ApimServiceName')]"
          }
        }
      },
      "name": "tagTemplate",
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2018-01-01",
      "dependsOn": []
    },
    {
      "properties": {
        "mode": "Incremental",
        "templateLink": {
          "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/gmsd1zu2amsplatfogene001-AccountsAPI-initial.api.template.json')]",
          "contentVersion": "1.0.0.0"
        },
        "parameters": {
          "ApimServiceName": {
            "value": "[parameters('ApimServiceName')]"
          }
        }
      },
      "name": "AccountsAPI-InitialAPITemplate",
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2018-01-01",
      "dependsOn": [
        "[resourceId('Microsoft.Resources/deployments', 'versionSetTemplate')]",
        "[resourceId('Microsoft.Resources/deployments', 'productsTemplate')]"
      ]
    },
    {
      "properties": {
        "mode": "Incremental",
        "templateLink": {
          "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/gmsd1zu2amsplatfogene001-AccountsAPI-subsequent.api.template.json')]",
          "contentVersion": "1.0.0.0"
        },
        "parameters": {
          "ApimServiceName": {
            "value": "[parameters('ApimServiceName')]"
          }
        }
      },
      "name": "AccountsAPI-SubsequentAPITemplate",
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2018-01-01",
      "dependsOn": [
        "[resourceId('Microsoft.Resources/deployments', 'AccountsAPI-InitialAPITemplate')]"
      ]
    }
  ]
}`

Here is the content of the parameters template file:

`{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "ApimServiceName": {
      "value": "gmsd1zu2amsplatfogene001"
    },
    "LinkedTemplatesBaseUrl": {
      "value": "https://gmsd1zu2staplatfogene012.blob.core.windows.net/api-management"
    }
  },
  "resources": []
}`

Thanks!

**Comment by RupengLiu** [comment]
![Annotation 2019-12-12 093848](https://user-images.githubusercontent.com/29927264/70735591-62cca280-1cc3-11ea-87a3-ed968b9c2c7e.png)
Hi, 
If you are using devops release to deploy these templates, can you help check if the area in the image has been set? 
Best,

**Comment by mfgonzalez** [comment]
Hi!
Thanks for your help! It think will help when I´ll be moving the APIs through the instances (environments like DEV, QA, UAT and PRD) but I also submitted a ticket to Azure support team and the original problem reported here was that I did not provide the "csmParametersFile" (or Template Parameters) value! I still having issues on my pipeline but I´ll close this issue since it was not a problem on the Creator tool!
Thanks!
